### ISSUE TITLE
[Feature Request] App access to API?

### ISSUE THREAD

**Issue by piotrkochan** [RocketChat/Rocket.Chat.Apps-engine#103]
I know that App only have access to it's Storage. However ability to fetch data from REST API (or maybe other?) would be a very powerful feature. App probably should be able to ask administrator for access to the data (like Android application) but before installation. 

Application is installed "locally" in RocketChat so calling REST API would be unnecessary. I believe there is a way to provide these data via other interface.

I want to open discussion about that feature, what are pros and cons, how hard it would be to implement that in Apps and on the RocketChat side.

**Comment by piotrkochan** [comment]
Ok, so plugin can communicate with RC API, still it is like a workaround. I think that Rocket.Chat.Apps-engine could provide some interface to provide access to RocketChat API data somehow better, HTTP calls will be not necessary.

**Comment by piotrkochan** [comment]
I don't have any usecase, I'm talking about that in general. Now I see `IUserBridge` and `IRoomBridge` interfaces. Are they implemented in the RC?

Also should not be there some way to allow application to access these data (during installation/app enabling)?

**Comment by d-gubert** [comment]
> Now I see `IUserBridge` and `IRoomBridge` interfaces. Are they implemented in the RC?

Yes, some of the code for the apps to work are in the Rocket.Chat repo. And here is an example on how you can read a user https://github.com/RocketChat/Apps.Atlassian.Jira/blob/000e0be6e898765cd13f9ecd3e809b1cef049f8a/slashcommands/jira.ts#L53

Also note you can't just query **all** the users/rooms/messasges from within the app, you can only get things by id.

> Also should not be there some way to allow application to access these data (during installation/app enabling)?

Yes, permissions are a work in progress for the engine

**Comment by piotrkochan** [comment]
> Yes, permissions are a work in progress for the engine

Could You provide some links?

> Also note you can't just query all the users/rooms/messasges from within the app, you can only get things by id.

Are there any plans to implement that?

**Comment by piotrkochan** [comment]
@d-gubert You said permissions are work in progress, did you mean #30 ? If yes, these permissions are about command to role availability related, not Plugin data access permissions

**Comment by d-gubert** [comment]
No, I wasn't talking about that specific issue. I don't have a link to provide about apps' permissions right now :/

> Are there any plans to implement that?

Not yet
